examples/js/loaders/GCodeLoader.js

```diff
@@ -0,0 +1,220 @@
+'use strict';
+
+/**
+ * THREE.GCodeLoader is used to load gcode files usually used for 3D printing or CNC applications.
+ *
+ * Gcode files are composed by commands used by machines to create objects.
+ *
+ * @class THREE.GCodeLoader
+ * @param {Manager} manager Loading manager.
+ * @author tentone
+ * @author joewalnes
+ */
+THREE.GCodeLoader = function ( manager ) {
+
+	this.manager = ( manager !== undefined ) ? manager : THREE.DefaultLoadingManager;
+
+};
+
+THREE.GCodeLoader.prototype.load = function ( url, onLoad, onProgress, onError ) {
+
+	var self = this;
+
+	var loader = new THREE.FileLoader( self.manager );
+	loader.load( url, function ( text ) {
+
+		onLoad( self.parse( text ) );
+
+	}, onProgress, onError );
+
+};
+
+THREE.GCodeLoader.prototype.parse = function ( data ) {
+
+	var currentState = { x: 0, y: 0, z: 0, e: 0, f: 0, extruding: false };
+	var currentLayer = undefined;
+	var relative = false;
+
+	var box = new THREE.Box3();
+	var layers = [];
+	
+	var pathMaterial = new THREE.LineBasicMaterial( { color: 0xFFFF00 } );
+	pathMaterial.name = 'path';
+
+	var extrudingMaterial = new THREE.LineBasicMaterial( { color: 0xFFFFFF } );
+	extrudingMaterial.name = 'extruded';
+
+	function newLayer( line ) {
+
+		currentLayer = { lines: [], currentLayer: layers.length, z: line.z };
+		layers.push( currentLayer );
+
+	}
+
+	function getLineGroup( line ) {
+
+		if ( currentLayer === undefined ) {
+
+			newLayer( line );
+
+		}
+
+		var grouptype = line.extruding ? 1 : 0;
+
+		if ( currentLayer.lines[ grouptype ] === undefined ) {
+
+			currentLayer.lines[ grouptype ] = {
+
+				type: grouptype,
+				feed: line.e,
+				extruding: line.extruding,
+				geometry: new THREE.Geometry(),
```

Comment:
BTW: It would be actually better to produce a `BufferGeometry` since we try to remove `Geometry` from the loaders.